### PR TITLE
[Merge only after UltraBrew changes are merged]Generate Metrics code in API Requests

### PR DESCRIPTION
**Pull request by OferLevi85**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java [hunk 1]
```diff
@@ -2536,15 +2433,14 @@ Access getAccessCheck(Principal principal, String action, String resource,
         // retrieve the domain based on our resource and action/trustDomain pair
 
         String domainName = retrieveResourceDomain(resource, action, trustDomain);
+        setRequestDomain(ctx, domainName);
         if (domainName == null) {
-            metric.increment(ZMSConsts.HTTP_REQUEST, ZMSConsts.ZMS_INVALID_DOMAIN, principal.getDomain());
-            metric.increment(caller, ZMSConsts.ZMS_INVALID_DOMAIN, principal.getDomain());
+            setRequestDomain(ctx, ZMSConsts.ZMS_INVALID_DOMAIN);
```
Comment (OferLevi85) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java:2438]: I'm setting the domain name here to "invalid_domain"

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java [hunk 2]
```diff
@@ -2536,15 +2433,14 @@ Access getAccessCheck(Principal principal, String action, String resource,
...
             throw ZMSUtils.notFoundError("getAccessCheck: Unable to extract resource domain", caller);
         }
         AthenzDomain domain = retrieveAccessDomain(domainName, principal);
         if (domain == null) {
-            metric.increment(ZMSConsts.HTTP_REQUEST, ZMSConsts.ZMS_UNKNOWN_DOMAIN, principal.getDomain());
-            metric.increment(caller, ZMSConsts.ZMS_UNKNOWN_DOMAIN, principal.getDomain());
+            setRequestDomain(ctx, ZMSConsts.ZMS_UNKNOWN_DOMAIN);
```
Comment (OferLevi85) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java:2443]: I'm setting the domain name here to "unknown_domain"

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java [hunk 3]
```diff
@@ -1054,12 +1054,7 @@ public DomainList getDomainList(ResourceContext ctx, Integer limit, String skip,
             String modifiedSince) {
 
         final String caller = "getdomainlist";
```
Comment (OferLevi85) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java:1056]: I originally had the caller (method name in lowercase) placed in the context in ZMSResources and then I was able to have the same line for every method: "final String caller = ctx.getApiName();".
The reason I didn't change it eventually as I noticed that it is sometimes used in authority checks (as the operation in verifyAuthorizedServiceOperation calls). So maybe it is best to keep it explicit.
Let me know what you think.

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java [hunk 4]
```diff
@@ -5385,6 +5026,7 @@ public Response getSignedDomains(ResourceContext ctx, String domainName, String
         if (domainName != null) {
             domainName = domainName.toLowerCase();
             validate(domainName, TYPE_DOMAIN_NAME, caller);
+            setRequestDomain(ctx, domainName);
```
Comment (OferLevi85) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java:5029]: Originally this method doesn't send the domain but I think it is correct to put it (if it exists)

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java [hunk 5]
```diff
@@ -5530,6 +5166,7 @@ public JWSDomain getJWSDomain(ResourceContext ctx, String domainName) {
         // policy, service, etc name)
 
         domainName = domainName.toLowerCase();
+        setRequestDomain(ctx, domainName);
```
Comment (OferLevi85) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java:5169]: Originally this method doesn't send the domain but I think it is correct to put it (if it exists)

File: rdl/rdl-gen-athenz-server/main.go [hunk 6]
```diff
@@ -407,8 +408,9 @@ func (gen *javaServerGenerator) resourcePath(r *rdl.Resource) string {
 
 func (gen *javaServerGenerator) handlerBody(r *rdl.Resource) string {
 	noContent := r.Expected == "NO_CONTENT" && r.Alternatives == nil
-	s := "        try {\n"
-	s += "            ResourceContext context = this.delegate.newResourceContext(this.request, this.response);\n"
+	s := "        int code = ResourceException.OK;\n"
+	s += "        ResourceContext context = this.delegate.newResourceContext(this.request, this.response);\n"
```
Comment (havetisyan) [inline rdl/rdl-gen-athenz-server/main.go:412]: Not sure we can move this call out of try block. If it throws an exception, it might expose our trace and some other internal data to the caller which we can't have. We should be able to define the ResourceContext context = null outside of the try block and keep the delegate call within the block and within the recordMetrics calls make sure to handle the case where context is null.

File: servers/zts/src/main/java/com/yahoo/athenz/zts/ZTSImpl.java [hunk 7]
```diff
@@ -2938,9 +2784,6 @@ public InstanceIdentity postInstanceRefreshInformation(ResourceContext ctx, Stri
             String domain, String service, String instanceId, InstanceRefreshInformation info) {
 
         final String caller = "postinstancerefreshinformation";
-        final String callerTiming = "postinstancerefreshinformation_timing";
-
-        metric.increment(HTTP_POST);
```
Comment (OferLevi85) [inline servers/zts/src/main/java/com/yahoo/athenz/zts/ZTSImpl.java:2943]: postInstanceRefreshInformation has many metrics that are not related to the HTTP request (such as "providerconfirm"). I didn't change them.
